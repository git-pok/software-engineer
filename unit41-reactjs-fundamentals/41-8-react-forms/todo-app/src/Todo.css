.Todo {
  display: flex;
  align-items: center;
}

.Todo-p {
  font-size: 25px;
  color: rgb(22, 21, 40);
  width: 600px;
}

.Todo-delete-icon, .Todo-edit-icon {
  width: 35px;
  height: 35px;
  font-size: 20px;
  font-weight: 900;
  margin: 20px 20px;
  background-color: rgb(22, 21, 40);
  color: rgb(237, 220, 169);
  border: 2px solid rgb(255, 255, 255);
  border-radius: 7px;
  transition: background-color .3s ease-in-out;
}

.Todo-edit-icon {
  width: 60px;
  height: 35px;
  font-size: 15px;
  font-weight: 900;
  margin: 20px 20px;
  background-color: rgb(22, 21, 40);
  color: rgb(237, 220, 169);
  border: 2px solid rgb(255, 255, 255);
  border-radius: 7px;
  transition: background-color .3s ease-in-out;
}

.Todo-delete-icon:hover, .Todo-edit-icon:hover {
  background-color: rgb(7, 203, 0);
  color: rgb(255, 0, 0);
}

.Todo-edit-icon:hover {
  background-color: rgb(22, 21, 40);
  color: rgb(255, 255, 255);
}