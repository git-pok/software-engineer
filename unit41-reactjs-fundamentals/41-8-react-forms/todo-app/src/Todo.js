import { useState } from 'react';
import EditTodoForm from './EditTodoForm.js';
import './Todo.css';

/**
 * Todo- this component should display a div with the task of the todo.
 */

const Todo = ({ todo, removeTodo, id, editTodo }) => {
  const [ isActive, setIsActive ] = useState(false);

  const editForm = () => {
    setIsActive(() => (
      !isActive      
    ));
  }

  return (
    <>
    <div className="Todo">
      <p className="Todo-p">{ todo }</p>
      <button
      className="Todo-delete-icon"
      onClick={() => removeTodo(id)}>X</button>
      <button
      className="Todo-edit-icon"
      onClick={editForm}
      >EDIT</button>
    </div>
    {
      isActive ? <EditTodoForm id={id} editTodo={editTodo} /> : null
    }
    </>
  );
}

export default Todo;