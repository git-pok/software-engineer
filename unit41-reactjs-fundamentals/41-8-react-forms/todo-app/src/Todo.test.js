import { render } from '@testing-library/react';
import {
  inputBc1, inputBc2, inputBc3,
  inputBcEdit1, inputBcEdit2
} from './_testCommon.js';
import Todo from './Todo';


test('renders <Todo /> without crashing', () => {
  render(<Todo />);
});


test('matches snapshot', () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});