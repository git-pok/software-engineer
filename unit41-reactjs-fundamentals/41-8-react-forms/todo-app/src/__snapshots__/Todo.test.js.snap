// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p
      class="Todo-p"
    />
    <button
      class="Todo-delete-icon"
    >
      X
    </button>
    <button
      class="Todo-edit-icon"
    >
      EDIT
    </button>
  </div>
</DocumentFragment>
`;
